tools/etcdhelper/etcdhelper.go

```diff
@@ -0,0 +1,113 @@
+package main
+
+import (
+	"crypto/tls"
+	"flag"
+	"fmt"
+	"os"
+
+	etcd "github.com/coreos/etcd/clientv3"
+	"golang.org/x/net/context"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/runtime/serializer/json"
+	"k8s.io/kubernetes/pkg/runtime/serializer/protobuf"
+
+	// install all APIs
+	_ "github.com/openshift/origin/pkg/api/install"
+	_ "github.com/openshift/origin/pkg/quota/api/install"
+	_ "k8s.io/kubernetes/pkg/api/install"
+)
+
+func main() {
+	var endpoint, keyFilePath, certFilePath string
+	flag.StringVar(&endpoint, "endpoint", "https://127.0.0.1:4001", "Etcd endpoint.")
+	flag.StringVar(&keyFilePath, "key-file", "", "TLS client key.")
+	flag.StringVar(&certFilePath, "cert-file", "", "TLS client certitificate.")
+	flag.Parse()
+
+	if flag.NArg() == 0 {
+		fmt.Fprintf(os.Stderr, "ERROR: you need to specify action ls [<key>] or get <key>\n")
+		os.Exit(1)
+	}
+	if flag.Arg(0) == "get" && flag.NArg() == 1 {
+		fmt.Fprintf(os.Stderr, "ERROR: you need to specify <key> for get operation\n")
+		os.Exit(1)
+	}
+	action := flag.Arg(0)
+	key := ""
+	if flag.NArg() > 1 {
+		key = flag.Arg(1)
+	}
+
+	cert, err := tls.LoadX509KeyPair(certFilePath, keyFilePath)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "ERROR: unable to read cert-file or key-file pair: %v\n", err)
+		os.Exit(1)
+	}
+
+	client, err := etcd.New(etcd.Config{
+		Endpoints: []string{endpoint},
+		TLS: &tls.Config{
+			Certificates:       []tls.Certificate{cert},
+			InsecureSkipVerify: true,
+		},
+	})
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "ERROR: unable to connect to etcd: %v\n", err)
+		os.Exit(1)
+	}
+	defer client.Close()
+
+	switch action {
+	case "ls":
+		if err = listKeys(client, key); err != nil {
+			fmt.Fprintf(os.Stderr, "ERROR: listing keys: %v\n", err)
+			os.Exit(1)
+		}
+	case "get":
+		if err = getKey(client, key); err != nil {
+			fmt.Fprintf(os.Stderr, "ERROR: getting %s: %v\n", key, err)
+			os.Exit(1)
+		}
+	}
+}
+
+func listKeys(client *etcd.Client, key string) error {
+	var resp *etcd.GetResponse
+	var err error
+	if len(key) == 0 {
+		resp, err = client.Get(context.TODO(), "/", etcd.WithFromKey(), etcd.WithKeysOnly())
+	} else {
+		resp, err = client.Get(context.TODO(), key, etcd.WithPrefix(), etcd.WithKeysOnly())
+	}
+	if err != nil {
+		return err
+	}
+	for _, ev := range resp.Kvs {
+		fmt.Printf("%s\n", ev.Key)
+	}
+	return nil
+}
+
+func getKey(client *etcd.Client, key string) error {
+	resp, err := client.Get(context.TODO(), key)
+	if err != nil {
+		return err
+	}
+	for _, ev := range resp.Kvs {
+		ps := protobuf.NewSerializer(api.Scheme, api.Scheme, "application/vnd.kubernetes.protobuf")
+		obj, gvk, err := ps.Decode(ev.Value, nil, nil)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "ERROR: unable to decode %s: %v\n", ev.Key, err)
+			continue
+		}
+		fmt.Println(gvk)
```

Comment:
This GVK, is the one that was read from etcd, afterwards I'm encoding this into json and print. I figured out this will be the easiest to work with.